lib/elixir_internal_certification/keyword/queries/keyword_lookup_query.ex

```diff
@@ -0,0 +1,341 @@
+        ^search_query
+      ) >= ^number_of_occurrences
+    )
+  end
+
+  defp condition_query(query, %AdvancedSearch{
+         search_query: search_query,
+         search_query_type: "occurrences",
+         search_query_target: "urls_of_non_adwords" = search_query_target,
+         symbol_notation: "<",
+         number_of_occurrences: number_of_occurrences
+       }) do
+    query
+    |> unnest_urls(search_query_target)
+    |> where(
+      [kl],
+      fragment(
+        "array_length(string_to_array(url_of_non_adwords, ?), 1) - 1",
+        ^search_query
+      ) < ^number_of_occurrences
+    )
+  end
+
+  defp condition_query(query, %AdvancedSearch{
+         search_query: search_query,
+         search_query_type: "occurrences",
+         search_query_target: "urls_of_non_adwords" = search_query_target,
+         symbol_notation: "<=",
+         number_of_occurrences: number_of_occurrences
+       }) do
+    query
+    |> unnest_urls(search_query_target)
+    |> where(
+      [kl],
+      fragment(
+        "array_length(string_to_array(url_of_non_adwords, ?), 1) - 1",
+        ^search_query
+      ) <= ^number_of_occurrences
+    )
+  end
+
+  defp condition_query(query, %AdvancedSearch{
+         search_query: search_query,
+         search_query_type: "occurrences",
+         search_query_target: "urls_of_non_adwords" = search_query_target,
+         symbol_notation: "=",
+         number_of_occurrences: number_of_occurrences
+       }) do
+    query
+    |> unnest_urls(search_query_target)
+    |> where(
+      [kl],
+      fragment(
+        "array_length(string_to_array(url_of_non_adwords, ?), 1) - 1",
+        ^search_query
+      ) == ^number_of_occurrences
+    )
+  end
+
+  defp unnest_urls(query, "urls_of_adwords_advertisers_top_position" = _search_query_target) do
+    query
+    |> with_cte("keyword_lookup_cte",
+      as:
+        fragment(
+          "select id, unnest(urls_of_adwords_advertisers_top_position) as url_of_adwords_advertisers_top_position, null as url_of_non_adwords from keyword_lookups"
+        )
+    )
+    |> join(:left, [kl], klc in "keyword_lookup_cte", on: klc.id == kl.id)
+    |> select_merge([kl], %{
+      url_of_adwords_advertisers_top_position: fragment("url_of_adwords_advertisers_top_position"),
+      url_of_non_adwords: fragment("url_of_non_adwords")
+    })
+  end
+
+  defp unnest_urls(query, "urls_of_non_adwords" = _search_query_target) do
+    query
+    |> with_cte("keyword_lookup_cte",
+      as:
+        fragment(
+          "select id, null as url_of_adwords_advertisers_top_position, unnest(urls_of_non_adwords) as url_of_non_adwords from keyword_lookups"
+        )
+    )
+    |> join(:left, [kl], klc in "keyword_lookup_cte", on: klc.id == kl.id)
+    |> select_merge([kl], %{
+      url_of_adwords_advertisers_top_position: fragment("url_of_adwords_advertisers_top_position"),
+      url_of_non_adwords: fragment("url_of_non_adwords")
+    })
+  end
```

Comment:
```suggestion
  defp unnest_urls(query) do
      query
      |> with_cte("keyword_lookup_cte",
        as:
          fragment(
            "select id, unnest(urls_of_adwords_advertisers_top_position) as url_of_adwords_advertisers_top_position, unnest(urls_of_non_adwords) as url_of_non_adwords from keyword_lookups"
          )
      )
      |> join(:left, [kl], klc in "keyword_lookup_cte", on: klc.id == kl.id)
      |> select_merge([kl], %{
        url_of_adwords_advertisers_top_position: fragment("url_of_adwords_advertisers_top_position"),
        url_of_non_adwords: fragment("url_of_non_adwords")
      })
    end
```

Can't we just have one function to unnest both types, ie result will be `null` for one column same as existing implementation?